docs/spec/metrics.md

```diff
@@ -31,6 +31,21 @@ If you want to track the metrics in Prometheus, you need to apply analogically
 PROMETHEUS=true
 ```
 
+For custom implemented endpoints, you need to wrap the endpoint with pod name mutator in order to make the endpoint pod name visible for the cross-connect monitor, who is responsible for exposing the metrics to Prometheus:
+```
+podName := endpoint.CreatePodNameMutator()
+composite := endpoint.NewCompositeEndpoint(
+	...,
+	endpoint.NewCustomFuncEndpoint("podName", podName),
+)
+```
```

Comment:
That is cool, we need to apply this to all examples here and in the examples repo.